pkg/operator/configobservation/apienablement/observe_runtime_config.go

```diff
@@ -14,9 +17,35 @@ import (
 	"github.com/openshift/library-go/pkg/operator/events"
 )
 
-var DefaultGroupVersionsByFeatureGate = map[configv1.FeatureGateName][]schema.GroupVersion{
-	"ValidatingAdmissionPolicy": {{Group: "admissionregistration.k8s.io", Version: "v1beta1"}},
-	"DynamicResourceAllocation": {{Group: "resource.k8s.io", Version: "v1alpha2"}},
+var defaultGroupVersionsByFeatureGate = map[configv1.FeatureGateName][]groupVersionByOpenshiftVersion{
+	"ValidatingAdmissionPolicy": {{GroupVersion: schema.GroupVersion{Group: "admissionregistration.k8s.io", Version: "v1beta1"}}},
+	"DynamicResourceAllocation": {
+		{To: ptr.To(semver.MustParse("1.31.0")), GroupVersion: schema.GroupVersion{Group: "resource.k8s.io", Version: "v1alpha2"}},
+		{From: ptr.To(semver.MustParse("1.31.0")), GroupVersion: schema.GroupVersion{Group: "resource.k8s.io", Version: "v1alpha3"}},
+	},
+}
+
+type groupVersionByOpenshiftVersion struct {
+	schema.GroupVersion
+	From *semver.Version
+	To   *semver.Version
```

Comment:
I see from the implementation that From is inclusive and To is exclusive. Could we use https://pkg.go.dev/github.com/blang/semver/v4#Range to model version ranges instead, and also verify that there is no intersection between ranges for a given API group (I think it would be fine to panic at package initialization time in that case rather than adding a test)?

Reply:
@benluddy yup Range fits this better and also added the intersection validation